### DISCUSSION TITLE
Installation issue of mtex-6.0.beta3 in macOS Sonoma Version 14.2.1

### DISCUSSION THREAD

**Discussion by aesirium** [mtex-toolbox/mtex#2083] [Ask Anything]
I have successfully installed mtex-6.0.beta2 before but with mtex-6.0.beta3, I keep getting this error:

MTEX: Couldn't get the mex files working!
   
The original error message was:
 
Error using check_installation>fast_check_mex
missing mex-file jcvoronoi_mex.mexmaca64

Error in check_installation (line 20)
    fast_check_mex

Error in startup_mtex (line 69)
check_installation;

Error in startup (line 9)
startup_mtex;

Can anyone please direct me on how to solve this?

**Comment by lfgmorales** [comment]
Hi

At the moment there is no clear solution for that unfortunately

There is a potential work around that somehow worked for me though, you can try to copy the jcvoronoi_mex.mexmaca64 file from the mex folder of your mtex(mtex—>mex) installation and paste it in mtex—>extern—>jcvoronoi

Restart matlab and try again, maybe that works

cheers

Luiz



Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.******@***.***>



On 4 Mar 2024, at 11:33, aesirium ***@***.******@***.***>> wrote:



I have successfully installed mtex-6.0.beta2 before but with mtex-6.0.beta3, I keep getting this error:

MTEX: Couldn't get the mex files working!

The original error message was:

Error using check_installation>fast_check_mex
missing mex-file jcvoronoi_mex.mexmaca64

Error in check_installation (line 20)
fast_check_mex

Error in startup_mtex (line 69)
check_installation;

Error in startup (line 9)
startup_mtex;

Can anyone please direct me on how to solve this?

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2083>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4LQKUCC3SBN2W6YE3YWREWVAVCNFSM6AAAAABEE565Y2VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGMYTSOBUGI>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2083", "url": "https://github.com/mtex-toolbox/mtex/discussions/2083", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by aesirium** [reply]
Thanks for the suggestion!

Unfortunately, I don't have the jcvoronoi_mex.mexmaca64 file in my 6.0.beta3 (mtex->mex) installation, just the jcvoronoi_mex.mexa64 and jcvoronoi_mex.mexw64.

**Comment by lfgmorales** [comment]
try this, not sure that works

cheers

Luiz


Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.******@***.***>



On 4 Mar 2024, at 11:52, aesirium ***@***.******@***.***>> wrote:



Thanks for the suggestion!

Unfortunately, I don't have the jcvoronoi_mex.mexmaca64 file in my 6.0.beta3 (mtex->mex) installation, just the jcvoronoi_mex.mexa64 and jcvoronoi_mex.mexw64.

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-8665132>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5INYWIOT65RIYTS4J4HDYWRG6VAVCNFSM6AAAAABEE565Y2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DMNRVGEZTE>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-8665132", "url": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-8665132", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

﻿ScanMail detected and quarantined a file named "jcvoronoi_mex.mexmaca64" that violated attachment blocking policy from the original mail entity. You can safely save or delete this replacement attachment.

**Reply by aesirium** [reply]
I can't access the attachment.

**Comment by lfgmorales** [comment]
try to download it from here

https://polybox.ethz.ch/index.php/s/7YFbOfFfgfJqw0j

cheers

Luiz

Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.******@***.***>



On 4 Mar 2024, at 12:06, aesirium ***@***.******@***.***>> wrote:



I can't access the attachment.

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-8665275>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5INZX3WUCHEPWEOQ6JWLYWRIRRAVCNFSM6AAAAABEE565Y2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DMNRVGI3TK>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-8665275", "url": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-8665275", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by aesirium** [reply]
Thank you! I was able to install it now.

**Reply by Tijmenvermeij** [reply]
Hi, 
I have the same problem on my intel Mac. I tried to download from this link but it was dead. Could you re-upload it maybe, @lfgmorales ?

**Comment by lfgmorales** [comment]
Hi

I have that as well, you can ignore it

let us know if things run

cheers


Luiz


Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.******@***.***>



On 4 Mar 2024, at 13:33, aesirium ***@***.******@***.***>> wrote:



Thank you! I was able to install it now and left with only this warning:

Warning: Function setappdata has the same name as a MATLAB built-in. We suggest you rename the function to avoid a potential name conflict.

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-8666177>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN5DJTVVHYYM6DE4OVTYWRSYXAVCNFSM6AAAAABEE565Y2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DMNRWGE3TO>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-8666177", "url": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-8666177", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by aesirium** [reply]
It's running now. Thanks for the immediate response!

**Comment by aesirium** [comment]
I was running some codes earlier and encountered a problem again with the jcvoronoi_mex.mexmaca64 file.
```
Invalid MEX-file '/Users/jb/Documents/MTEX/mtex-6.0.beta3/mex/jcvoronoi_mex.mexmaca64':
dlopen(/Users/jb/Documents/MTEX/mtex-6.0.beta3/mex/jcvoronoi_mex.mexmaca64, 0x0006): tried:
'/Users/jb/Documents/MTEX/mtex-6.0.beta3/mex/jcvoronoi_mex.mexmaca64' (not a mach-o file)
```

**Comment by lfgmorales** [comment]
What are you trying to do?

Calculate grains?

If so, either you use the ‘qhull’ option every time you calculate grains, or more easily, you go to the mtex_settings.m and change the line to 56 to

setMTEXpref('voronoiMethod','qhull’);

restart matlab

the problem is that jcvoronoi is not properly compiled yet to mtex in macs

cheers


Luiz



Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.******@***.***>



On 5 Mar 2024, at 11:39, aesirium ***@***.******@***.***>> wrote:



I was running some codes earlier and encountered a problem again with the jcvoronoi_mex.mexmaca64 file.

Invalid MEX-file '/Users/jb/Documents/MTEX/mtex-6.0.beta3/mex/jcvoronoi_mex.mexmaca64':
dlopen(/Users/jb/Documents/MTEX/mtex-6.0.beta3/mex/jcvoronoi_mex.mexmaca64, 0x0006): tried:
'/Users/jb/Documents/MTEX/mtex-6.0.beta3/mex/jcvoronoi_mex.mexmaca64' (not a mach-o file)

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-8678511>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN7DHYSQDABNMFW46GTYWWOFVAVCNFSM6AAAAABEE565Y2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DMNZYGUYTC>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-8678511", "url": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-8678511", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by aesirium** [reply]
Thank you! It's running well again.

**Comment by lfgmorales** [comment]
Hi Tijmen

can you try this file (you need to remove the txt additional extension I have used

cheers

Luiz

Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.***



On 17 Apr 2024, at 11:35, Tijmen Vermeij ***@***.***> wrote:



Hi,
I have the same problem on my intel Mac. I tried to download from this link but it was dead. Could you re-upload it maybe, @lfgmorales<https://github.com/lfgmorales> ?

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-9140228>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN2OCTU346UBNRYUBP3Y5Y67HAVCNFSM6AAAAABEE565Y2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TCNBQGIZDQ>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-9140228", "url": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-9140228", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]


﻿ScanMail detected and quarantined a file named "jcvoronoi_mex.mexmaca64.txt" that violated attachment blocking policy from the original mail entity. You can safely save or delete this replacement attachment.

**Reply by Tijmenvermeij** [reply]
Hi, it seems Github removes your attachment... I think you may need to reply through the Github website and upload the file there...
Thanks

**Comment by lfgmorales** [comment]
Hi Tijmen

Can you try this and see if it works?

<https://polybox.ethz.ch/index.php/s/iSLflTExzMmRsG8>
[favicon-fb.png]
polybox - <a href="https://polyboxdoc.ethz.ch/" target="_blank">polybox Help</a><br>Use it as a logical memory stick<br>Store your data on the ETH campus<https://polybox.ethz.ch/index.php/s/iSLflTExzMmRsG8>
polybox.ethz.ch<https://polybox.ethz.ch/index.php/s/iSLflTExzMmRsG8>




cheers

Luiz


Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.***



On 17 Apr 2024, at 11:35, Tijmen Vermeij ***@***.***> wrote:



Hi,
I have the same problem on my intel Mac. I tried to download from this link but it was dead. Could you re-upload it maybe, @lfgmorales<https://github.com/lfgmorales> ?

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-9140228>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN2OCTU346UBNRYUBP3Y5Y67HAVCNFSM6AAAAABEE565Y2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TCNBQGIZDQ>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-9140228", "url": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-9140228", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by lfgmorales** [comment]
I uploaded on the eth “dropbox” and sent you another email with a link, did you get it?

cheers

Luiz



Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.***



On 17 Apr 2024, at 13:50, Tijmen Vermeij ***@***.***> wrote:



Hi, it seems Github removes your attachment... I think you may need to reply through the Github website and upload the file there...
Thanks

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-9141692>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5INYZ2IA6F5XRBPUGMB3Y5ZOW7AVCNFSM6AAAAABEE565Y2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TCNBRGY4TE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-9141692", "url": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-9141692", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by Tijmenvermeij** [reply]
Hi, yes I got it.
I tried to put this file both in the mex folder and in the extern/jcvoronoi folder, but it doesn't work. Isn't this file you send, `jcvoronoi_mex.mexmaca64` meant for the new ARM based Macs?
Mtex tells me: 
```
Error using check_installation>fast_check_mex (line 115)
missing mex-file jcvoronoi_mex.mexmaci64
```

**Comment by lfgmorales** [comment]
Oh, sorry, I didnt read your message properly

I will upload the one I have for mexmaci64 now, same folder

cheers

Luiz



Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.***



On 17 Apr 2024, at 13:56, Tijmen Vermeij ***@***.***> wrote:



Hi, yes I got it.
I tried to put this file both in the mex folder and in the extern/jcvoronoi folder, but it doesn't work. Isn't this file you send, jcvoronoi_mex.mexmaca64 meant for the new ARM based Macs?
Mtex tells me:

Error using check_installation>fast_check_mex (line 115)
missing mex-file jcvoronoi_mex.mexmaci64


—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-9141810>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN43R7XXGGLJBN2N4E3Y5ZPODAVCNFSM6AAAAABEE565Y2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TCNBRHAYTA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-9141810", "url": "https://github.com/mtex-toolbox/mtex/discussions/2083#discussioncomment-9141810", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by Tijmenvermeij** [reply]
Thanks, I tried it and got the following error when using grainCalc:

```
>> grains = calcGrains(ebsd)
Invalid MEX-file '/Users/tvermeij/Documents/MATLAB/mtex-5.11.1/mex/jcvoronoi_mex.mexmaci64':
dlopen(/Users/tvermeij/Documents/MATLAB/mtex-5.11.1/mex/jcvoronoi_mex.mexmaci64, 0x0006): tried:
'/Users/tvermeij/Documents/MATLAB/mtex-5.11.1/mex/jcvoronoi_mex.mexmaci64' (not a mach-o file),
'/System/Volumes/Preboot/Cryptexes/OS/Users/tvermeij/Documents/MATLAB/mtex-5.11.1/mex/jcvoronoi_mex.mexmaci64' (no such file),
'/Users/tvermeij/Documents/MATLAB/mtex-5.11.1/mex/jcvoronoi_mex.mexmaci64' (not a mach-o file)

Error in spatialDecomposition (line 35)
      [Vx,Vy,E1,E2,I_ED1,I_ED2] = jcvoronoi_mex([X;dummyCoordinates]);

Error in EBSD/calcGrains (line 70)
  [V,F,I_FD] = spatialDecomposition([ebsd.prop.x(:), ebsd.prop.y(:)],ebsd.unitCell,varargin{:});
 
```
